### PR TITLE
Issue-535 Fixed owerlap when some files uploaded

### PR DESCRIPTION
**Pull request by michael-volynets**
#535

### CODE REVIEW

**Comment by tomholub** [comment]
Thanks! I will review it. For issues that change dexign or UX, please include a screen shot after the change in the PR. That will be super helpful (in the future).

**Comment by michael-volynets** [comment]
Okay, Got it,

image

**Comment by tomholub** [comment]
I just realized what the issue is, and here is how I resolved it before:
```js
  /**
   * On Firefox, we have to manage textbox height manually. Only applies to composing new messages
   * (else ff will keep expanding body element beyond frame view)
   * A decade old firefox bug is the culprit: https://bugzilla.mozilla.org/show_bug.cgi?id=202081
   *
   * @param updateRefBodyHeight - set to true to take a new snapshot of intended html body height
   */
  private setInputTextHeightManuallyIfNeeded = (updateRefBodyHeight: boolean = false) => {
    if (!this.urlParams.isReplyBox && Catch.browser().name === 'firefox') {
      let cellHeightExceptText = 0;
      for (const cell of this.S.cached('all_cells_except_text')) {
        cellHeightExceptText += $(cell).is(':visible') ? ($(cell).parent('tr').height() || 0) + 1 : 0; // add a 1px border height for each table row
      }
      if (updateRefBodyHeight || !this.refBodyHeight) {
        this.refBodyHeight = this.S.cached('body').height() || 605;
      }
      this.S.cached('input_text').css('height', this.refBodyHeight - cellHeightExceptText);
    }
}
```
You'll just need to update that function. I'm guessing at the end you will subtract attachment box height also, and make sure this function is also called when attachments are added or removed.

**Comment by tomholub** [comment]
For the attachments themselves, it would be better if the the attachments themselves had variable width based on the file name (with some maximum). Something like this:

image

Now that I'm looking at it, it would be great if the `X` was also centered with the borders: 

image

Also the space between attachments should be the same vertically as horizontally:

image

Thanks!

**Comment by michael-volynets** [comment]
The result after all Changes:
image

Checked on Firefox and it works fine

File: extension/js/common/composer.ts [hunk 1]
```diff
@@ -1543,7 +1544,7 @@ export class Composer {
       }
     }, this.getErrHandlers(`focus on recipient field`))).children().click(() => false);
     this.resizeInputTo();
-    this.attach.initAttDialog('fineuploader', 'fineuploader_button');
+    this.attach.initAttDialog('fineuploader', 'fineuploader_button', async () => this.setInputTextHeightManuallyIfNeeded());
```
Comment (tomholub) [inline extension/js/common/composer.ts]: Would it be possible to re-use `this.attach.setAttAddedCb(...)` ? That way we would not need to add a new `onSuccess` callback above.

File: extension/js/common/browser.ts [hunk 2]
```diff
@@ -1266,6 +1266,7 @@ export class AttUI {
         return;
       }
       this.attachedFiles[uploadFileId] = newFile;
+      onSuccess!().catch(Catch.reportErr);
```
Comment (tomholub) [inline extension/js/common/browser.ts]: You will get exception if `onSuccess` was not defined (see other note, adding callback may not be necessary).

File: extension/js/common/composer.ts (hunk 1 above)
Comment (michael-volynets) [reply extension/js/common/composer.ts]: oh, yes, I didn't see that method.

**Comment by tomholub** [comment]
Normally you don't need to test on Firefox too much. It's mostly design changes that can come out a bit unexpected on FF, and this one was an especially tricky one, where the behavior on FF is very different. Normally this won't be an issue.

**Comment by michael-volynets** [comment]
Ok, got it

**Pull request merged**